XenAdmin/Dialogs/OptionsPages/ExternalToolsOptionsPage.cs

```diff
@@ -40,72 +40,96 @@ namespace XenAdmin.Dialogs.OptionsPages
             var puttyPath = Properties.Settings.Default.PuttyLocation;
             var openSshPath = Properties.Settings.Default.OpenSSHLocation;
-            if (radioButtonPutty.Checked && !string.IsNullOrEmpty(puttyPath) && puttyPath == textBoxPutty.Text ||
-                radioButtonOpenSsh.Checked && !string.IsNullOrEmpty(openSshPath) && openSshPath == textBoxOpenSsh.Text)
+
+            if ((selectedConsole == SshConsole.Putty && radioButtonPutty.Checked ||
+                 selectedConsole == SshConsole.OpenSSH && radioButtonOpenSsh.Checked) &&
+                puttyPath == textBoxPutty.Text &&
+                openSshPath == textBoxOpenSsh.Text)
             {
                 return true;
             }
 
-            return !(radioButtonPutty.Checked && !File.Exists(textBoxPutty.Text) ||
-                     radioButtonOpenSsh.Checked && !File.Exists(textBoxOpenSsh.Text) ||
-                     textBoxPutty.Text.Length > 0 && !textBoxPutty.Text.ToLower().EndsWith(".exe") ||
-                     textBoxOpenSsh.Text.Length > 0 && !textBoxOpenSsh.Text.ToLower().EndsWith(".exe"));
-        }
-
-        public void ShowValidationMessages()
-        {
-            var message = string.Empty;
-            if (textBoxPutty.Text.Length > 0 && !textBoxPutty.Text.ToLower().EndsWith(".exe"))
-            {
-                _tooltipControl = textBoxPutty;
-                message = Messages.EXTERNAl_TOOLS_FILE_INVALID;
-            }
-            else if (textBoxOpenSsh.Text.Length > 0 && !textBoxOpenSsh.Text.ToLower().EndsWith(".exe"))
-            {
-                _tooltipControl = textBoxOpenSsh;
-                message = Messages.EXTERNAl_TOOLS_FILE_INVALID;
-            }
-            else if (radioButtonPutty.Checked && !File.Exists(textBoxPutty.Text))
+            if (radioButtonPutty.Checked)
             {
-                _tooltipControl = textBoxPutty;
-                message = Messages.FILE_NOT_FOUND;
+                if (string.IsNullOrEmpty(textBoxPutty.Text) || !textBoxPutty.Text.ToLower().EndsWith(".exe"))
+                {
+                    _tooltipControl = textBoxPutty;
+                    tooltipValidation.ToolTipTitle = Messages.EXTERNAL_TOOLS_FILE_INVALID;
+                    return false;
+                }
+
+                if (!File.Exists(textBoxPutty.Text))
+                {
+                    _tooltipControl = textBoxPutty;
+                    tooltipValidation.ToolTipTitle = Messages.FILE_NOT_FOUND;
+                    return false;
+                }
             }
-            else if (radioButtonOpenSsh.Checked && !File.Exists(textBoxOpenSsh.Text))
+            else if (radioButtonOpenSsh.Checked)
             {
-                _tooltipControl = textBoxOpenSsh;
-                message = Messages.FILE_NOT_FOUND;
+                if (string.IsNullOrEmpty(textBoxPutty.Text) || !textBoxOpenSsh.Text.ToLower().EndsWith(".exe"))

```

Comment:
First check in the `if` should contain `textBoxOpenSsh.Text`

Reply:
🤦